src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -722,6 +735,42 @@ private IConversionExpression CreateBoundAsOperatorOperation(BoundAsOperator bou
             return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
+        private IDelegateCreationExpression CreateBoundDelegateCreationExpressionOperation(BoundDelegateCreationExpression boundDelegateCreationExpression)
+        {
+            Lazy<IOperation> target = new Lazy<IOperation>(() =>
+            {
+                if (boundDelegateCreationExpression.Argument.Kind == BoundKind.MethodGroup &&
+                    boundDelegateCreationExpression.MethodOpt != null)
+                {
+                    // If this is a method binding, and a valid candidate method was found, then we want to expose
+                    // this child as an IMethodBindingReference. Otherwise, we want to just delegate to the standard
+                    // CSharpOperationFactory behavior
+                    BoundMethodGroup boundMethodGroup = (BoundMethodGroup)boundDelegateCreationExpression.Argument;
+                    return CreateBoundMethodGroupSingleMethodOperation(boundMethodGroup, boundDelegateCreationExpression.MethodOpt, boundMethodGroup.SuppressVirtualCalls);
+                }
+                else
+                {
+                    return Create(boundDelegateCreationExpression.Argument);
+                }
+            });
+            SyntaxNode syntax = boundDelegateCreationExpression.Syntax;
+            ITypeSymbol type = boundDelegateCreationExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundDelegateCreationExpression.ConstantValue);
+            bool isImplicit = boundDelegateCreationExpression.WasCompilerGenerated;
+            return new LazyDelegateCreationExpression(target, _semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private IMethodBindingExpression CreateBoundMethodGroupSingleMethodOperation(BoundMethodGroup boundMethodGroup, IMethodSymbol methodSymbol, bool suppressVirtualCalls)
+        {
+            bool isVirtual = methodSymbol != null && (methodSymbol.IsAbstract || methodSymbol.IsOverride || methodSymbol.IsVirtual) && !suppressVirtualCalls;
```

Comment:
>methodSymbol != null [](start = 29, length = 20)

Do we ever expect methodSymbol to be null here?

Reply:
I don't believe we do. Fixing.

---
In reply to: [140097698](https://github.com/dotnet/roslyn/pull/22056#discussion_r140097698) [](ancestors = 140097698)